frontend/src/components/VErrorSection/VErrorImage.vue

```diff
@@ -38,7 +38,9 @@ export default defineComponent({
      * the code of the error, used to identify and render the appropriate image
      */
     errorCode: {
-      type: String as PropType<ErrorCode>,
+      type: String as PropType<
+        Extract<ErrorCode, "NO_RESULT" | "SERVER_TIMEOUT">
+      >,
```

Comment:
Why is `Exctract` needed for this? No result and server timeout are the only two error codes, right? Is there one added in a follow up PR that causes this requirement?

One kind of weird side-effect of using `Extract` is that it won't complain if something on the right-hand side isn't actually part of the left-hand side: https://www.typescriptlang.org/play?#code/C4TwDgpgBAogTnA9nAwogJtAvFARAOQHkB9AJRgGUBVAGQBVcoAfPCmUgNXeLoEkBZGISoMAUKNCQodKDhgAPYHACGAY2AAeeElQYIAGjxEylWg2at2XUjwFCRjFgUIAFGLgB8ogPTeo-gD0AflEgA

Preferably there'd be some kind of error that said "Hey, `NOPE` isn't ever part of `ErrorCode`!". Seeing as we're already defining the two we want inline anyway, I'd vote for us to import the two consts and do something like:

```suggestion
      type: String as PropType<
        typeof NO_RESULT
        | typeof SERVER_TIMEOUT
      >,
```

Or if we really want to make sure that `ErrorCode` is part of it, use `Exclude` instead of `Extract` so that we don't run into the issue where an ErrorCode is removed or changed but isn't handled.

I prefer just defining the union inline. If we really need to use something like `Extract`, a custom type would be able to cause a type error when the members don't extend the union: https://www.typescriptlang.org/play?#code/C4TwDgpgBAogTnA9nAwogJtAvFARAOQHkB9AJRgGUBVAGQBVcoAfPCmUgNXeLoEkBZGISoMAUKNCQoFAK4AjAM4RgAHioA7AJaJ1UCAA9gEdegVQFwOJvUBzADRR+EALZyIcMwaMmzG7eoA+KBwYQzgAQwBjVT8dBydXdwUA8UloOmDpeSVVeCRUDAgHAhJyanpGFlw2Tm4+QWEGZjwiAAUYXBSAei6oPoA9AH5xIA

Anyway, not really an issue, just a nit because this looks confusing and tempting for someone to remove given the 1-to-1 relationship between the left and right side of this right now. If that's changing in the future, all good, but a comment explaining why we can't define the accepted error codes inline and need to use extract instead would be helpful.

Reply:
Thank you for the detailed context! I replaced the type in this PR with the simple union, and will include the suggestions in #2838, which actually changes the `FetchingError` type.